### PR TITLE
Parse merchant container and pass to localstorage

### PR DESCRIPTION
**Pull request by M-OC**
- Parses the container returned by tagmanager into a simpler format. (see "parseContainer")
- Creates utility methods for reading/writing "ContainerSummary" from local storage.
- Updates tests to account for "pageView" triggering requests on load. (tests were previously hard-coded to expect '0', '1' requests. These values have been modified +1)
- Adds tests/mocks for new "fetchContainerSettings" method.

### CODE REVIEW

File: src/lib/local-storage/container.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  return null;
+};
+
+/* Returns an existing, non-expired container. Removes a container
+from localstorage if it has expired. */
+export const getValidContainer = () : ContainerSummary | null => {
+  const storedValue = getContainer();
+  const now = Date.now();
+
+  if (!storedValue || ((now - storedValue.createdAt) > oneHour)) {
```
Comment (joshbeam) [inline src/lib/local-storage/container.js:23]: I'm wondering if we can do even lower than an hour. I think the considerations we have to balance are:
1. Allowing merchants to send as many offers as possible
2. Latency of getting the container

What would be the effect if we dropped this down to say, 15 minutes?

**Comment by joshbeam** [comment]
Can you rename this PR to something more descriptive?

File: src/lib/get-property-id.js [hunk 2]
```diff
@@ -3,11 +3,32 @@ import { getMerchantID } from '@paypal/sdk-client/src';
...
+  ContainerSummary
 } from '../types';
 
-import { getPropertyId, setPropertyId } from './local-storage';
+import { getPropertyId, setPropertyId, setContainer, getValidContainer } from './local-storage';
 
+/* Takes the full container and transforms it into
+a format better suited for use by the SDK */
+const parseContainer = (container : Container) : ContainerSummary => {
+  const offerTag = container.tags.find(tag => tag.tag_definition_id === 'offers');
```
Comment (joshbeam) [inline src/lib/get-property-id.js]: Internet Explorer doesn't support `Array.prototype.find` :( 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: test/get-property-id.test.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  describe('fetchContainerSettings', () => {
+    beforeEach(() => {
+      window.localStorage.clear();
+    });
+
+    it('requests a container using merchantId and browser location', async (done) => {
+      const expectedUrl = 'https://www.paypal.com/tagmanager/containers/xo?mrid=759SBALRW3ZTY&url=http%3A%2F%2Flocalhost';
+      await fetchContainerSettings({});
+
+      expect(global.fetch).toHaveBeenCalledTimes(1);
```
Comment (joshbeam) [inline test/get-property-id.test.js:37]: do you need this, if you're testing `toHaveBeenCalledWith` below?

File: src/lib/local-storage/container.js (hunk 1 above)
Comment (M-OC) [reply src/lib/local-storage/container.js:23]: This is a complete guess, but I'd wager you'd see a noticeable increase in the number of requests made to tagmanager.

File: src/lib/get-property-id.js (hunk 2 above)
Comment (M-OC) [reply src/lib/get-property-id.js]: I think Babel _should_ be transpiling polyfills for us? I'll do some manual testing on IE before next Wednesday. (that assumption might be wrong.)

Screen Shot 2019-10-09 at 2 31 52 PM

Comment (M-OC) [reply src/lib/get-property-id.js]: Came across this in paypal-checkout-components:
Screen Shot 2019-10-09 at 2 41 04 PM

File: test/get-property-id.test.js (hunk 3 above)
Comment (M-OC) [reply test/get-property-id.test.js:37]: Just double-checked. It's necessary: "toHaveBeenCalledWith" doesn't care about how many times a method is called.

File: src/lib/get-property-id.js (hunk 2 above)
Comment (M-OC) [reply src/lib/get-property-id.js]: Updated.

**Pull request merged**